MapleServer2/PacketHandlers/Game/UserChatHandler.cs

```diff
@@ -169,18 +194,82 @@ private static void HandleWhisperChat(GameSession session, string recipient, str
                 return;
             }
 
+            if (itemLinkPacket != null)
+            {
+                recipientPlayer.Session.Send(itemLinkPacket);
+                session.Send(itemLinkPacket);
+            }
             recipientPlayer.Session.Send(ChatPacket.Send(session.Player, message, ChatType.WhisperFrom));
             session.Send(ChatPacket.Send(recipientPlayer, message, ChatType.WhisperTo));
         }
 
-        private static void HandleClubChat(/*GameSession session, string message, ChatType type, long clubId*/)
+        private static void HandleClubChat(/*GameSession session, string message, ChatType type, long clubId, Packet itemLinkPacket*/)
         {
             // TODO
         }
 
-        private static void HandleChat(GameSession session, string message, ChatType type)
+        private static void HandleChat(GameSession session, string message, ChatType type, Packet itemLinkPacket)
         {
+            if (itemLinkPacket != null)
+            {
+                session.FieldManager.BroadcastPacket(itemLinkPacket);
+            }
             session.FieldManager.SendChat(session.Player, message, type);
         }
+
+        private static Packet GetItemLink(string message)
+        {
+            // '<' signals a message containing an item link
+            if (!message.Contains('<'))
+            {
+                return null;
+            }
+            Packet itemLinkPacket = null;
+
+            Match a = Regex.Match(message, @"<A (.*?)>");
+            string[] itemLinkMessages = Regex.Matches(message, @"<A (.*?)>")
```

Comment:
Since it appears that HTML is being parsed here, you should instead use an HTML or XML parser. The emulator already makes use of the latter

Reply:
the only part being html here is that the link is enclosed in a html <a> tag. 
the <a is not relevant by itself but it ensures people can't fake item links via chat messages, as the < is escaped by the client if typed in chat.
I don't want to spend more time on item linking, it works and I'm not sure what the benefit would be using html parser here

Reply:
I guess I could use the htm parser for the one line in question here, but I don't know how to use it and I don't have energy left for item linking.

Reply:
Not sure why you insist on being hesitant to improve your code, but the benefits are that you can dramatically improve readability by using a parser. Sure, your code works, but code simply working shouldn't be the minimum acceptance criteria. I legitimately could not tell what your code was trying to parse, especially on line 238, and I had to ask in Discord for an example. 

Here's some sample code that is shorter, much more readable (even if you had no idea what the input looked like, this code basically documents what it is trying to parse for you), and more robust since we are offloading the parsing to a battle-tested native library rather than writing custom regex:

```
        private static void GetItemLink(string message)
        {
            XmlDocument doc = new XmlDocument();
            doc.LoadXml("<xml>" + message + "</xml>");
            foreach (XmlNode link in doc.SelectNodes("//A"))
            {
                string[] hrefSplit = link.Attributes["HREF"].Value.Split(",");
                string itemLinkType = hrefSplit[0].Split(":")[1];
                long itemUid = long.Parse(hrefSplit[1]);
            }
        }
```
Parsing what is clearly HTML/XML using regex is universally considered a bad idea for many reasons.

Reply:
gotta agree with alan using the html parser or even the xml parser which already exists is a better option. Also we might only have this example that users html tags for now but in the future we might find more.

Reply:
The code didn't work as is and I am not sure of what is going wrong. 
My main point is I want to move on to another feature. 
I think my code is fine and readable. 
If someone looks at the variable names it's very understandable what's going on.
Since the system here has multiple layers to get the information of the item link, of course one needs to take the time to understand each layer (in TERMS of the variables  that are being created).
Trying to understand exactly what each string manipulation is doing is completely unnecessary.
It's like someone would look at every functioon used in another function in order to understand what is going on in the function those functions are called in.
The string manipulations do not need to be understood in order to understand the code, and there are as many instructions with the parser, actually even more when creating the parser instance. Therefore I think they are equally understandable.
All someone needs to understand is, "a regex is being parsed here and assigned to this variable"
no need to understand the content of the regex.
Don't get me wrong I really appreciate the input from you, I learned a lot by that.

While this code is html syntax we don't care that is html, we don't need any of the html information.
We could use another part of the string to match with, it would also work.
I only do it to avoid people creating a fake item message by typing the syntax of the item link in chat, as the server gets the item link info from the message and there is no mode or other information which makes the item link message unique. The only thing that makes it unique is the "<" character, which is escaped by the client to "&lt" so that is the only way to make sure it was send by the client.

A regex should be more performant here as well, no need of all the class overhead of the parser.
It is in my opinion as readable as the other.

I am not resistant to improving my code, but I don't want to make that many revisions, when the change is so simple and as I tried to explain, in my perspective, has little benefit if at all. 

and while reading html by regexs would probably be bad in html contexts, I think "reading html" is the problem here, we could as well read from : to , by a regex or use other string manipulation. 

Lastly, I want to move on to other features and not spend 2 days of my free time on 25 lines of code (the research took the most time of course - and testing) The code works, and I think the syntax is understandable. 
I tried changing it to the xml parser EVEN though I dislike it compared to my code, but it gave me errors.
So, if someone wants to change it to using an xml/html parser/, they can do it, although I think it's unneccessary. I will continue with another feature or so from now on.

Reply:
While arbitrary HTML with just a regex is a bad idea and almost impossible, it's sometimes appropriate to use them for parsing a limited, known set of HTML. In this case, you have a small set of HTML tags that you want to extract data from the package, regular expressions can work fine. Sometimes we have to know the scope to be used, and this is just a particular case.